akka-distributed-data/src/main/scala/akka/cluster/ddata/ORSet.scala

```diff
@@ -148,6 +207,21 @@ final class ORSet[A] private[akka] (
 
   type T = ORSet[A]
 
+  // Optimization for add/remove followed by merge, which is very common
+  // and merge should just fast forward to the new set
+  private var ancestor: ORSet[A] = null
```

Comment:
It's actually a cache between calls of the same thread, you can think of it as a thread local.
The Replicator actor invokes the user's modify function, which returns a new ORSet instance, with the ancestor field set, see for example the `add` method. Then (in same thread) the Replication calls `merge`, which makes use of the `ancestor` field to perform quick merge.